### PR TITLE
docs: initial migration and troubleshooting guides

### PR DESCRIPTION
**Pull request by beastawakens**

### CODE REVIEW

File: docs/help/troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+And to retry your installation after a few minutes.  If your Rack consistently fails to install into your Cloud provider and there are no relevant issues reported then please raise a [support ticket](/help/support) for us to investigate.
+
+### Scaling
+
+Scaling issues can often arise when you run into quota limits set by your cloud provider.  Ensure you have adequate limits in place for your scaling needs by contacting your cloud provider's support to get them set appropriately.
+
+## Running your Apps
+
+### Healthchecks
```
Comment (ddollar) [inline docs/help/troubleshooting.md]: Health checks should be two words here and in the prose below

File: docs/help/troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+### Installation
+
+One of the majoy advantages of Convox is that it abstracts you from the complexities involved in installing Kubernetes clusters on the cloud provider of your choice.  You are still subject, unfortunately, to any errors or issues that your provider is experiencing.  If encountering an error during installation, it's worth checking the Status page for your provider:
+
+- [AWS](https://status.aws.amazon.com/)
+- [GCP](https://status.cloud.google.com/)
+- [Azure](https://status.azure.com/en-us/status)
+- [Digital Ocean](https://status.digitalocean.com/)
+
+And to retry your installation after a few minutes.  If your Rack consistently fails to install into your Cloud provider and there are no relevant issues reported then please raise a [support ticket](/help/support) for us to investigate.
```
Comment (ddollar) [inline docs/help/troubleshooting.md]: Note on how to retry, `convox rack update -r <rackname>`

File: docs/help/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Troubleshooting
+
+## Local Rack
```
Comment (ddollar) [inline docs/help/troubleshooting.md]: Development Rack

File: docs/help/troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+### Installation
+
+Ensure you have followed the setup instructions for your local OS [here](../installation/development-rack).  Memory requirements for running Docker/Kubernetes locally can often catch people out.
+
+## Cloud Rack
```
Comment (ddollar) [inline docs/help/troubleshooting.md]: Production Rack

File: docs/help/troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+- [AWS](https://status.aws.amazon.com/)
+- [GCP](https://status.cloud.google.com/)
+- [Azure](https://status.azure.com/en-us/status)
+- [Digital Ocean](https://status.digitalocean.com/)
+
+And to retry your installation after a few minutes.  If your Rack consistently fails to install into your Cloud provider and there are no relevant issues reported then please raise a [support ticket](/help/support) for us to investigate.
+
+### Scaling
+
+Scaling issues can often arise when you run into quota limits set by your cloud provider.  Ensure you have adequate limits in place for your scaling needs by contacting your cloud provider's support to get them set appropriately.
```
Comment (ddollar) [inline docs/help/troubleshooting.md]: How do they know if this is happening?

File: docs/help/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+## Running your Apps
+
+### Healthchecks
+
+Any services within your app that expose a port will require a passing health check before receiving traffic.  Deploying a Release of your App that does not pass the health checks will result in a rollback to the previous release.  If this is your first release of a new app, a failing health check will result in a failed deployment.
```
Comment (camerondgray) [inline docs/help/troubleshooting.md]: you should add a link to: /configuration/health-checks

File: docs/help/migration.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# Migrating from previous Convox racks
+
+## Local Racks
+
+- To retain your local Apps when moving to the new local Rack, you should [export](../reference/cli/apps#apps-export) them all first.  This will create a local archive of all pertinent data for each app you export.
```
Comment (ddollar) [inline docs/help/migration.md]: Should probably note that they need to do the uninstall with the old CLI

**Comment by scottmessinger** [comment]
I followed these directions on two of our developers computers and both got this:

```
Error: clusterroles.rbac.authorization.k8s.io "atom" already exists

  on .terraform/modules/system/terraform/api/k8s/atom.tf line 1, in resource "kubernetes_cluster_role" "atom":
   1: resource "kubernetes_cluster_role" "atom" {

Error: clusterrolebindings.rbac.authorization.k8s.io "atom" already exists

  on .terraform/modules/system/terraform/api/k8s/atom.tf line 13, in resource "kubernetes_cluster_role_binding" "atom":
  13: resource "kubernetes_cluster_role_binding" "atom" {

Error: clusterroles.rbac.authorization.k8s.io "router" already exists

  on .terraform/modules/system/terraform/router/k8s/main.tf line 5, in resource "kubernetes_cluster_role" "router":
   5: resource "kubernetes_cluster_role" "router" {

Error: Waiting for service "dev-system/router" to assign IP/hostname for a load balancer

  on .terraform/modules/system/terraform/router/local/main.tf line 40, in resource "kubernetes_service" "router":
  40: resource "kubernetes_service" "router" {
```

It appears resetting k8s helps solve it, but frankly, I got it working on one machine and am still working to get it working on the other, so I'm not entirely sure.

**Pull request merged**